[IMP] accounting/l10n_ec: add eCommerce section (#8522)
odoo/documentation: content/applications/finance/fiscal_localizations/ecuador.rst

   withholding of 70% in a new line with the same tax support, the system will allow you as long as
   the total of the bases matches the total from the :guilabel:`Vendor Bill`.

eCommerce
---------

> This :ref:`ATS Report <ecuador/ats>` module enables the features and configurations to:

Author: @xpl-odoo I agree with this suggestion.

> - Choose the SRI Payment Method in each payment method configuration.

Author: @xpl-odoo agree on suggestion.

- Customers can manually input their identification type and identification number in the checkout

>   process on the website.

Author: @xpl-odoo agree on suggestion.

- Automatically generate a valid electronic invoice for Ecuador at the end of the checkout process.

Configuration
~~~~~~~~~~~~~

The following configurations are needed to effectively generate electronic invoices after the

> checkout process on the website.

Author: @xpl-odoo I agree with the suggestion.

> Website settings

Author: @xpl-odoo I agree with the suggestion.

****************

To enable automatic invoicing, navigate to :menuselection:`Website --> Configuration --> Settings:
Invoicing` section and activate the option :guilabel:`Automatic Invoice`. With this option active,
an invoice will be generated after the checkout process.

.. tip::
   The email template for the invoice can be modified from the :guilabel:`Invoice Email Template`
   field under the :guilabel:`Automatic Invoice` option.

.. important::
   The sales journal used for invoicing will be the first in the sequence of priority in the
   :guilabel:`Journal` menu.

Payment providers
*****************

Activate the payment providers that will be used to capture payments through the eCommerce website.
To do so, navigate to :menuselection:`Website --> Configuration --> Settings: Shop - Payment`
section and then click on the :guilabel:`View other providers` button under the :guilabel:`Activate
Payments` heading. From here, each payment provider can be configured by selecting a provider
record. Once a provider is configured, be sure to click the :guilabel:`Publish` button so it is
available in the checkout process when paying.

Payment methods
***************

> Activate the payment methods that each payment provider can have. To do this, click on the option
> :guilabel:`Enable Payment Methods` button within the :guilabel:`Configuration` tab of each provider.
> This will allow each provider to have one or more payment methods.

Reviewer: ```suggestion
To activate one or more payment methods for a payment provider, click
:guilabel:`→ Enable Payment Methods` within the :guilabel:`Configuration` tab of each provider.

```

When configuring the payment method, it is **mandatory** to set the :guilabel:`SRI Payment Method`

> for each method. This field will appear after you create and save the payment method for the first
> time.

Reviewer: ```suggestion
for each method. This field appears after you create and save the payment method for the first
time.
```

Author: @xpl-odoo I agree with the suggestion.

.. note::

>    The :guilabel:`SRI Payment Method` field will not be visible when configuring the payment method
>    for the first time. This information is mandatory to correctly generate the electronic invoice
>    from the eCommerce sale. To access that field, click on a **payment method** to access its
>    configuration menu.

Reviewer: ```suggestion
   Adding the :guilabel:`SRI Payment Method` is necesary to generate correctly the electronic invoice
   from an eCommerce sale. Select a **payment method** to access its configuration menu and the field.
```
I'm not sure this should be in a note

Author: @xpl-odoo Yes, it should be noted.

Reviewer: I meant that it could be part of the core text; I wasn't sure using the note markup was relevant :)

.. seealso::
   :doc:`../payment_providers`

.. image:: ecuador/l10n-ec-sri-payment-method.png
   :align: center
   :alt: l10n_ec SRI Payment Method.

eCommerce workflow
~~~~~~~~~~~~~~~~~~

Type and identification number
******************************

> The client who is making a purchase will have the option in the checkout process to indicate their
> identification type and number. This information is required to correctly generate the electronic

Reviewer: ```suggestion
The client who is making a purchase will have the option to indicate their
identification type and number during the checkout process. This information is required to correctly generate the electronic
```

Author: @xpl-odoo Good catch.

invoice after the checkout is completed.

.. image:: ecuador/website-checkout-form.png
   :align: center
   :alt: Website checkout form.

.. note::

>    A verification exists to verify that the *Identification Number* field is completed and has the
>    correct number of digits. For RUC identification, 13 digits are required. For Cédula, 9 digits
>    are required.

Reviewer: ```suggestion
   Verification is done to ensure the *Identification Number* field is completed and has the
   correct number of digits. For RUC identification, 13 digits are required. For Cédula, 9 digits
   are required.
```

Author: @xpl-odoo I agree with the suggestion.

> After finishing the checkout process, a confirmed invoice is generated ready to be sent manually or

Reviewer: ```suggestion
After finishing the checkout process, a confirmed invoice is generated, ready to be sent manually or
```

Author: @xpl-odoo I agree with the suggestion.

  eCommerce checkout process.
- Automatically generate a valid electronic invoice for Ecuador at the end of the checkout process.

Configuration
~~~~~~~~~~~~~

Website
*******

To generate an invoice after the checkout process, navigate to :menuselection:`Website -->
Configuration --> Settings` and activate the :guilabel:`Automatic Invoice` option found under the
:guilabel:`Invoicing` section.

.. tip::
   The invoice's email template can be modified from the :guilabel:`Invoice Email Template`.

.. important::
   The sales journal used for invoicing is the first in the sequence of priority in the
   :guilabel:`Journal` menu.

Payment providers
*****************

To activate the payment providers that should be used to capture eCommerce payments, navigate to
:menuselection:`Website --> Configuration --> Payment Providers` section and then click on the
:guilabel:`View other providers` button under the :guilabel:`Activate Payments` heading. From here,
each payment provider can be configured by selecting a provider record. Refer to the :doc:`payment
provider <../payment_providers>` documentation for more information.

Payment methods

> ~~~~~~~~~~~~~~~

Reviewer: ```suggestion
^^^^^^^^^^^^^^^
```
My bad, I used the wrong character to make the suggestion. I wanted this to be at the H6 level :)

To activate one or more payment methods for a payment provider, click :guilabel:`→ Enable Payment
Methods` within the :guilabel:`Configuration` tab of each provider.

When configuring the payment method, it is **mandatory** to set the :guilabel:`SRI Payment Method`
for each method. This field appears after you create and save the payment method for the first
time.

.. note::
   Adding the :guilabel:`SRI Payment Method` is necessary to generate correctly the electronic
   invoice from an eCommerce sale. Select a **payment method** to access its configuration menu and
   the field.

.. seealso::

>    :doc:`../payment_providers`

Reviewer: I would label this "payment provider" as on lines 605-606 or the other way around, just to be consistent
